kentikapi/users_integration_test.go

```diff
@@ -194,6 +195,108 @@ func TestClient_GetAllUsers(t *testing.T) {
 	}
 }
 
+func TestClient_GetUser_WithTimeout(t *testing.T) {
+	tests := []struct {
+		name            string
+		responses       []testutil.HTTPResponse
+		expectedResult  *models.User
+		expectedError   bool
+		expectedTimeout bool
+		// timeout and expected requests are strictly connected. The server sleeps for 1 ms before each response.
+		timeout      *time.Duration
+		expectedReqs int
+	}{
+		{
+			name: "default timeout",
+			responses: []testutil.HTTPResponse{
+				testutil.NewErrorHTTPResponse(http.StatusBadGateway),
+				testutil.NewErrorHTTPResponse(http.StatusBadGateway),
+				{StatusCode: http.StatusBadRequest, Body: `{"error":"Bad Request"}`},
+			},
+			expectedError:   true,
+			expectedTimeout: false,
+			expectedReqs:    3,
+		}, {
+			name: "status bad request",
+			responses: []testutil.HTTPResponse{
+				{StatusCode: http.StatusBadRequest, Body: `{"error":"Bad Request"}`},
+			},
+			expectedError:   true,
+			expectedTimeout: false,
+			timeout:         durationPtr(1 * time.Second),
+			expectedReqs:    1,
+		}, {
+			name: "timeout is longer than the wait for response with retries",
+			responses: []testutil.HTTPResponse{
+				testutil.NewErrorHTTPResponse(http.StatusBadGateway),
+				testutil.NewErrorHTTPResponse(http.StatusBadGateway),
+				{StatusCode: http.StatusBadRequest, Body: `{"error":"Bad Request"}`},
+			},
+			expectedError:   true,
+			expectedTimeout: false,
+			timeout:         durationPtr(10 * time.Second),
+			expectedReqs:    3,
+		}, {
+			name: "timeout after 1 request",
```

Comment:
```suggestion
			name: "timeout during first request",
```
First request won't even finish, so "after" is not precise

Reply:
Done